libphx/src/Window.cpp

```diff
@@ -77,6 +82,29 @@ void Window_SetVsync (Window*, bool vsync) {
   SDL_GL_SetSwapInterval(vsync ? 1 : 0);
 }
 
+void Window_SetCursor(Window* self, cstr name, int hotx, int hoty) {
+  //self->cursor = SDL_CreateSystemCursor(SDL_SystemCursor::SDL_SYSTEM_CURSOR_CROSSHAIR);
+
+  SDL_FreeCursor(self->cursor);   // Can take NULL
+  
+  cstr path = Resource_GetPath(ResourceType_Tex2D, name);
+
+  int width, height, components;
+  uchar* data = Tex2D_LoadRaw(path, &width, &height, &components);
+
+  SDL_PixelFormatEnum format = (components == 3) ? SDL_PIXELFORMAT_RGB24 : SDL_PIXELFORMAT_RGBA32;
+  SDL_Surface* surface = SDL_CreateRGBSurfaceWithFormatFrom(data, width, height, components * 8, width * components, format);
+  if (!surface)
+    Fatal("Failed to create custom cursor surface for window");
+
+  self->cursor = SDL_CreateColorCursor(surface, hotx, hoty);
+  if (!self->cursor)
+    Fatal("Failed to create custom cursor for window");
```

Comment:
This will leak the surface object in this scenario, could you add `SDL_FreeSurface` to this block?